### PR TITLE
Use spec defined basePath when available.

### PR DESCRIPTION
**Pull request by nikhilm**

### CODE REVIEW

**Comment by nikhilm** [comment]
Apologies for not adding tests in the original PR. Will update this soon.

File: generator/support.go [hunk 1]
```diff
@@ -671,7 +671,7 @@ func (a *appGenerator) makeCodegenApp() (GenApp, error) {
 
 	basePath := "/"
 	if sw.BasePath != "" {
-		basePath = "/"
+		basePath = basePath
```
Comment (casualjim) [inline generator/support.go]: this should be sw.BasePath no?

Comment (nikhilm) [inline generator/support.go]: Sorry, between github's outage and something I was working on, I'm clearly not thinking straight.

Comment (casualjim) [inline generator/support.go]: no worries that's what PR's are for :)

**Pull request closed without merging**